### PR TITLE
[REF] models: separate ui plugins in three concepts

### PR DESCRIPTION
**Pull request by pro-odoo**
Before this commit, plugins were split in two: core and ui, based on whether a state should be shared in collaborative.

With this commit, we split the ui plugins in three categories:
- Plugins which have a state, but which should not be shared in collaborative (edition, selection) => stateful_ui_plugins
- Plugins which have a derived state from core data (evaluation) => core_views_plugins
- Plugins which handle a specific feature, without handling any core commands (autofill, sort) => feature_plugins

For now, the only difference is that a command replayed will be dispatched to core view plugins (typically to ensure the evaluation is correct even after a command replay).

But in the future, we should give the history to stateful ui plugins, in order to save the selection or the active sheet when a core command is dispatched.

Part of task-id 2813749

## Description:

description of this task, what is implemented and why it is implemented that way.

Odoo task ID : [TASK_ID](https://www.odoo.com/web#id=TASK_ID&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -298,7 +317,10 @@ export class Model extends EventBus<any> implements CommandDispatcher {
         this.state.recordChanges.bind(this.state),
         (command: CoreCommand) => {
           this.isReplayingCommand = true;
-          this.dispatchToHandlers([this.range, ...this.corePlugins], command);
+          this.dispatchToHandlers(
+            [this.range, ...this.corePlugins, ...this.coreViewsPlugins],
```
Comment (LucasLefevre) [inline src/model.ts:321]: this can be tested:
A core command dispatching an `UPDATE_CELL` command should invalidate evaluation

**Review by LucasLefevre — APPROVED** [review]

File: src/model.ts (hunk 1 above)
Comment (pro-odoo) [reply src/model.ts:321]: Done: https://github.com/odoo/o-spreadsheet/pull/1817/files#diff-cf5735e699ad89c54032c224139a1cca3f60889cd5d5d30381d59043e3f47a63R201

File: src/plugins/index.ts [hunk 2]
```diff
@@ -46,26 +50,30 @@ export const corePluginRegistry = new Registry<CorePluginConstructor>()
...
   .add("autofill", AutofillPlugin)
   .add("find_and_replace", FindAndReplacePlugin)
   .add("sort", SortPlugin)
   .add("automatic_sum", AutomaticSumPlugin)
   .add("format", FormatPlugin)
   .add("cell_popovers", CellPopoverPlugin)
-  .add("selection_multiuser", SelectionMultiUserPlugin)
+  .add("selection_multiuser", SelectionMultiUserPlugin);
+
+export const statefulUIPluginRegistry = new Registry<UIPluginConstructor>()
```
Comment (LucasLefevre) [inline src/plugins/index.ts:68]: faut updater la doc :/

**Comment by rrahir** [comment]
@robodoo r+

**Pull request closed without merging**